server/expression/array.go

```diff
@@ -153,33 +152,26 @@ func (array *Array) WithResolvedChildren(children []any) (any, error) {
 		}
 		newExpressions[i] = resolvedExpression
 	}
-	return &Array{
-		children:    newExpressions,
-		coercedType: array.coercedType,
-	}, nil
+	return array.WithChildren(newExpressions...)
 }
 
 // getTargetType returns the evaluated type for this expression. Returns the "anyarray" type if the type combination is
 // invalid.
-func (array *Array) getTargetType() pgtypes.DoltgresArrayType {
-	if array.coercedType != nil {
-		return array.coercedType
-	}
+func getTargetType(children ...sql.Expression) (pgtypes.DoltgresArrayType, error) {
```

Comment:
This is more specific to the `Array` expression. It doesn't necessarily use the `(array *Array)` parameter, but making it a method of the struct frees up the global namespace a bit.